### DISCUSSION TITLE
Optimization - Flexible Budget (Goal Based)

### DISCUSSION THREAD

**Discussion by annie-gulati** [google/meridian#244] [Q&A] [answered]
In the meridian.analysis.optimizer.BudgetOptimizer function, there is an option to specify
target ROI or target MROI to perform goal based optimization. Similarly, is there any feature to define a
target revenue / KPI which will enable us to perform goal based optimization by directly specifying the revenue or KPI value instead of ROI or MROI?

**Comment by mikewurm** [comment, marked as answer]
Thank you for the question. If I understand correctly, you have a target expected revenue and would like to achieve this expected revenue while minimizing the overall media spend. This is not currently implemented; however, it might be possible to solve this with an iterative approach.

- Run a fixed budget optimization with total budget B.
- If the optimized expected revenue (ROI * B) exceeds target revenue, then decrease B and try again. Conversely, if optimized expected revenue falls short of target, then increase B and try again.
